client/payment-details/types.ts

```diff
@@ -9,6 +9,7 @@ export interface PaymentChargeDetailsResponse {
 	data: PaymentIntent | Charge;
 	error: ApiError;
 	isLoading: boolean;
+	doRefund: ( charge: Charge, reason: string | null ) => void;
```

Comment:
@mgascam @KarlisJ , this looks a bit weird to me - we don't have `doCapture`, but adding `doRefund`. Any insights, on why we don't have the same approach?

Reply:
@kalessil I'll try my best answer based on my understanding but I may be off. The reason why `doCapture` is not there is because the types for the `useAuthorization` hook are defined inline here https://github.com/Automattic/woocommerce-payments/blob/cd45dcb7b620a28d187d9a340c02e59cb3df2d9e/client/data/authorizations/hooks.ts#L79.  IMO the `doCapture` does not belong in the `PaymentChargeDetailsResponse` type because we are talking about two different hooks that are interested in two different parts of the state: PaymentIntents and Charges on one hand and Authorizations on the other.

Having said that, there may be an opportunity to extract the inline types definition of the `useAuthorization` hook into its own interface, same as the `PaymentChargeDetailsResponse`. But truth is it is only used in one place at the moment, so I don't see the point. 

Furthermore, the usage of `PaymentChargeDetailsResponse` in https://github.com/Automattic/woocommerce-payments/blob/e00a4af880a9fa9d1d764e2b9ffe93b8ada10e26/client/payment-details/charge-details/index.tsx#L31 and https://github.com/Automattic/woocommerce-payments/blob/34eb13c4b140cebd233314eb06d4daa750ae86fb/client/payment-details/summary/refund-modal/index.tsx#L42 seem redundant to me, because the TS compiler should already able to infer the type based on the return type of the `usePaymentIntentWithChargeFallback` hook.  In my understanding of TS, the 'as' keyword should be used sparingly, since you are essentially telling the compiler to trust the human. 

Hope to clarify things, please let me know if you have any question.

Reply:
Yes, that gives more clarity. Thanks!

Since there's not much we can do with generalization here, marking the threa as resolved.